### PR TITLE
Pod setup restart fixes and improvements (Loop issue #2117)

### PR DESCRIPTION
**Pull request by itsmojo**

### CODE REVIEW

**Comment by marionbarker** [comment]
Successful test with Eros pod.

Please see OmniBLE test notes as well at: https://github.com/LoopKit/OmniBLE/pull/113#issuecomment-1935182660

Tested with Eros pod using an SE phone running iOS 17.3.
Configuration:

* LoopWorkspace-dev, commit 7340d55
* OmniBLE, dev branch commit 8f78c34
* OmniKit, main branch commit f2fc341
* Modified Pod.swift cannulaInsertionUnitsExtra from 0.0 to 5.0 to make the insertion process last longer.

Test during Pair/Priming:
* Touch Pair and wait for pod to begin priming
* quit Loop app, restart Loop app, touch the Finish Setup icon on main screen
   * pod was still priming, returns to Pair Pod screen, reconnects and shows Priming progress bar (repeated 2 times)
* quit Loop app, restart Loop app, touch the Finish Setup icon on main screen
   * pod was finished priming, returns to Pair Pod screen, reconnects and shows the check Paired with Continue button available

Test autolock disable
* wait 30 sec between taps when moving through screens to verify phone does not lock

Test during insertion
* Code was modified to insert 5.5 units before completing insertion
   * So insertion will last about 110 sec
* start insertion, wait for clicking and progress bar
* quit Loop app, restart Loop app, touch the Finish Setup icon on main screen
   * pod was still giving insertion bolus (clicking), returns to Insert Cannula screen, reconnects and shows Inserting progress bar (repeated 2 times)
* quit Loop app, restart Loop app, touch the Finish Setup icon on main screen
   * pod was finished with insertion bolus, returns to Insert Cannula screen, reconnects and shows the check Inserted with Continue button available

**Comment by marionbarker** [comment]
Code review:
* Compared code for last commit of this PR with code for last commit of OmniBLE 113
* Code changes are consistent

See testing performed with DASH pod at https://github.com/LoopKit/OmniBLE/pull/113#issuecomment-1947555014

File: OmniKit/PumpManager/PodCommsSession.swift [hunk 1]
```diff
@@ -435,7 +435,8 @@ public class PodCommsSession {
             if status.podProgressStatus == .insertingCannula {
                 podState.setupProgress = .cannulaInserting
                 podState.updateFromStatusResponse(status, at: currentDate)
-                return insertionWait // Not sure when it started, wait full time to be sure
```
Comment (ps2) [inline OmniKit/PumpManager/PodCommsSession.swift:438]: Is `insertionWait` no longer used? Maybe it should be removed above.

File: OmniKitUI/ViewControllers/OmnipodUICoordinator.swift [hunk 2]
```diff
@@ -214,6 +214,11 @@ class OmnipodUICoordinator: UINavigationController, PumpManagerOnboarding, Compl
                 self?.navigateTo(.deactivate)
             }
 
+            if pumpManager.podCommState == .activating {
```
Comment (ps2) [inline OmniKitUI/ViewControllers/OmnipodUICoordinator.swift]: This logic seems like view model logic. Can you expose the needed api `pumpManager.podCommState` from the `PodPairer` protocol, and move this logic into the viewmodel?

File: OmniKitUI/ViewModels/InsertCannulaViewModel.swift [hunk 3]
```diff
@@ -134,6 +135,13 @@ class InsertCannulaViewModel: ObservableObject, Identifiable {
         }
     }
     
+    var restarting: Bool {
```
Comment (ps2) [inline OmniKitUI/ViewModels/InsertCannulaViewModel.swift]: I found this name "restarting" to be confusing, as I couldn't figure out what was being restarted. Is "resuming" maybe a more descriptive name? They can't actually restart pairing.  They're just trying to resume where they left off, right?

File: OmniKitUI/Views/InsertCannulaView.swift [hunk 4]
```diff
@@ -76,9 +76,13 @@ struct InsertCannulaView: View {
             .padding()
         }
         .alert(isPresented: $cancelModalIsPresented) { cancelPairingModal }
-        .navigationBarTitle("Insert Cannula", displayMode: .automatic)
+        .navigationBarTitle(LocalizedString("Insert Cannula", comment: "navigation bar title for insert cannula"), displayMode: .automatic)
         .navigationBarBackButtonHidden(true)
         .navigationBarItems(trailing: cancelButton)
+        .onFirstAppear {
+            // handle possible restart without waiting for a button action
+            self.viewModel.handlePossibleRestart()
```
Comment (ps2) [inline OmniKitUI/Views/InsertCannulaView.swift]: This also should be view model logic, and not a view triggered action. That will obviate the need for the custom `onFirstAppear` ViewModifier.

**Review by ps2 — CHANGES_REQUESTED** [review]
This is an important fix, thanks.  It just seems like some of the view model logic is misplaced in the navigator and views.

File: OmniKit/PumpManager/PodCommsSession.swift (hunk 1 above)
Comment (itsmojo) [reply OmniKit/PumpManager/PodCommsSession.swift:438]: Removed insertionWait and replaced its one remaining use with a calculation based on status return

File: OmniKitUI/ViewControllers/OmnipodUICoordinator.swift (hunk 2 above)
Comment (itsmojo) [reply OmniKitUI/ViewControllers/OmnipodUICoordinator.swift]: Done using your suggested updates

File: OmniKitUI/Views/InsertCannulaView.swift (hunk 4 above)
Comment (itsmojo) [reply OmniKitUI/Views/InsertCannulaView.swift]: Done using your suggested updates

File: OmniKitUI/ViewModels/InsertCannulaViewModel.swift (hunk 3 above)
Comment (itsmojo) [reply OmniKitUI/ViewModels/InsertCannulaViewModel.swift]: This is all gone now using your suggested updates

**Pull request merged**